tests/raftstore/node.rs

```diff
@@ -16,35 +16,64 @@
 use std::collections::{HashMap, HashSet};
 use std::sync::{Arc, RwLock};
 use std::time::Duration;
+use std::fs::File;
+use std::io;
 
 use rocksdb::DB;
+use tempdir::TempDir;
+use protobuf::Message;
 
 use super::cluster::{Simulator, Cluster};
 use tikv::server::Node;
-use tikv::raftstore::store::{self, Transport, msg, SendCh};
+use tikv::raftstore::store::*;
 use kvproto::raft_cmdpb::*;
 use kvproto::raft_serverpb;
+use kvproto::raftpb::MessageType;
 use tikv::raftstore::Result;
 use tikv::util::HandyRwLock;
 use tikv::server::Config as ServerConfig;
 use tikv::server::transport::{ServerRaftStoreRouter, RaftStoreRouter};
+use tikv::util::codec::number::NumberEncoder;
 use super::pd::TestPdClient;
 use super::transport_simulate::{SimulateTransport, Filter};
 
 pub struct ChannelTransport {
-    pub routers: HashMap<u64, Arc<RwLock<ServerRaftStoreRouter>>>,
+    snap_paths: HashMap<u64, TempDir>,
+    routers: HashMap<u64, Arc<RwLock<ServerRaftStoreRouter>>>,
 }
 
 impl ChannelTransport {
     pub fn new() -> Arc<RwLock<ChannelTransport>> {
-        Arc::new(RwLock::new(ChannelTransport { routers: HashMap::new() }))
+        Arc::new(RwLock::new(ChannelTransport {
+            snap_paths: HashMap::new(),
+            routers: HashMap::new(),
+        }))
     }
 }
 
 impl Transport for ChannelTransport {
     fn send(&self, msg: raft_serverpb::RaftMessage) -> Result<()> {
+        let from_store = msg.get_from_peer().get_store_id();
         let to_store = msg.get_to_peer().get_store_id();
 
+        if msg.get_message().get_msg_type() == MessageType::MsgSnapshot {
+            let snap = msg.get_message().get_snapshot();
+            let source_file = match self.snap_paths.get(&from_store) {
+                Some(p) => SnapFile::from_snap(p.path(), snap, SNAP_GEN_PREFIX).unwrap(),
+                None => return Err(box_err!("missing temp dir for store {}", from_store)),
+            };
+            let mut dst_file = match self.snap_paths.get(&to_store) {
+                Some(p) => SnapFile::from_snap(p.path(), snap, SNAP_REV_PREFIX).unwrap(),
+                None => return Err(box_err!("missing temp dir for store {}", to_store)),
+            };
+            let mut reader = File::open(source_file.path()).unwrap();
+
+            dst_file.encode_u64(msg.compute_size() as u64).unwrap();
```

Comment:
No, the snapshot file to be sent and the file finally saved to disk are not identical.